Editorial pass on use of the word requires (#348)
w3c/wcag2ict: success-criteria-problematic-for-closed-functionality.md

Other success criteria would apply to systems with closed functionality either if they are partially closed or if they allow for the connection of some types of devices. As an example, Success Criterion 2.1.1 Keyboard would apply to systems which are closed to screen readers, but have a physical keyboard or a connector for standard keyboards.  While these criteria, as written, are not always applicable to closed functionality, most of them can inform and aid development of built-in features needed to make products with closed functionality accessible. 

> For non-web software on products closed functionality, alternate accessibility provisions might be needed to cover the user needs addressed by the following success criteria:

Reviewer: ```suggestion
For non-web software on products with closed functionality, alternate accessibility provisions might be needed to cover the user needs addressed by the following success criteria:
```

<ul>
<li><a href="#non-text-content">1.1.1 Non-text Content</a> — Depends upon text (or a text alternative) being in a programmatically determinable form.</li>
<li><a href="#audio-only-and-video-only-prerecorded">1.2.1 Pre-recorded video</a> — One of the options available to authors for success criterion 1.2.1 is providing a media alternative that is text which, in the absence of connected assistive technology, would need to be made available in different modalities.</li>
<li><a href="#audio-description-or-media-alternative-prerecorded">1.2.3 Audio description or Media Alternative</a> — One of the options available to authors for success criterion 1.2.3 is providing a media alternative that is text which, in the absence of connected assistive technology, would need to be made available in different modalities.</li>
<li><a href="#info-and-relationships">1.3.1 Info and Relationships</a> — Requires that information be in a programmatically determinable form or in text (that is programmatically determinable).</li>

> <li><a href="#meaningful-sequence">1.3.2 Meaningful Sequence</a> — Requires information (i.e, a correct reading sequent) in a programmatically determinable form.  An eqquivalent for software with closed functionality would be to provide a meaningful reading sequence through auditory output or some other non-visual means that helps users correlate the output with the corresponding information displayed on the screen.</li>

Reviewer: ```suggestion
<li><a href="#meaningful-sequence">1.3.2 Meaningful Sequence</a> — Requires information (i.e, a correct reading sequence) in a programmatically determinable form.  An equivalent for software with closed functionality would be to provide a meaningful reading sequence through auditory output or some other non-visual means that helps users correlate the output with the corresponding information displayed on the screen.</li>
```

<li><a href="#orientation">1.3.4 Orientation</a> — Products with closed functionality that have fixed-in-place displays or other limitations to modifying the physical display orientation are covered under the essential exception and are not required to provide support for orientation changes.  See the note in the section <a href="#applying-sc-1-3-4-orientation-to-non-web-documents-and-software">Applying SC 1.3.4 Identify Input Purpose to Non-Web Documents and Software</a>.</li>
<li><a href="#identify-input-purpose">1.3.5 Identify Input Purpose</a> — Depends upon information in a programmatically determinable form; in the absence of programmatic capabilities, text labels need to be specific and be provided to the user in other modalities (e.g. auditory).</li>

> <li><a href="#audio-control">1.4.2 Audio Control</a> — The intent of this success criterion is to avoid interference of audio with assistive products, which are not available in a system with closed functionality.  If the built-in accessibility features of a closed system provide speech output, then the interference may happen and this SC applies.  In addition, there are existing requirements in regulation (e.g., the EN 301 549 and U.S. Revised 508 Standards) that address volume control for products with closed functionality.</li>

Reviewer: ```suggestion
<li><a href="#audio-control">1.4.2 Audio Control</a> — The intent of this success criterion is to avoid interference of audio with assistive products, which are not available in a system with closed functionality.  If the built-in accessibility features of a closed system provide speech output, then the interference may happen and this SC applies.  In addition, there are existing requirements in regulations (e.g., the EN 301 549 and U.S. Revised 508 Standards) that address volume control for products with closed functionality.</li>
```

<div class="note">Photographs (e.g., of a hardware display) are not sufficient for testing that content meets this success criterion.  This is because the quality of the lighting, camera, and physical aspects of the hardware display can dramatically affect the ability to capture the content for testing purposes.</div></li>
</ul></li>
<li><a href="#resize-text">1.4.4 Resize Text</a> — Because the text rendering support in a closed environment may be more limited than the support found in user agents for the Web, meeting success criterion 1.4.4 in a closed environment may place a much heavier burden on the content author;</li>
<li><a href="#images-of-text">1.4.5 Images of Text</a> — To enable assistive technology to modify displayed text (e.g., adjusting contrast, increasing font size), machine-readable text is needed, as opposed to mere images of text.  Not all ICT with closed functionality has the capability to support visual modification of displayed text or images of text, given there is no interoperability with assistive technology and/or lack of platform support.</li>

> <li><a href="#reflow">1.4.10 Reflow</a> — Many procucts with closed functionality do not allow users to modify the viewport or change font sizes, so there would be no need to impose a requirement on such product that content be able to reflow.  Additionally, many products with closed functionality do not display large chunks of text and only have UI controls; in such cases, two-dimensional scrolling to access the text and UI controls may be considered essential.</li>

Reviewer: ```suggestion
<li><a href="#reflow">1.4.10 Reflow</a> — Some software on ICT with closed functionality does not support scrolling content or changing the viewport (examples include software for self-service transaction machines or kiosks). Therefore, there would be no content to which this success criterion applies, meaning the success criterion would be satisfied.  Instead, other requirements exist outside of WCAG to address use by persons with low vision (including but not limited to using sufficiently large text and single screen designs).

<div class="note">NOTE: Some ICT with closed functionality does not display large chunks of text and only has UI controls.  In such cases, two-dimensional scrolling to access the text and UI controls may be considered essential, thus meeting an exception, and the success criterion would be satisfied.</div></li>
```

<li><a href="#pointer-cancellation">2.5.2 Pointer Cancellation</a> — As noted in the section <a href="#applying-sc-2-5-2-pointer-cancellation-to-non-web-documents-and-software">Applying SC 2.5.2 Pointer Cancellation to Non-Web Documents and Software</a>, examples of ‘essential’ functionality are features for meeting environmental energy usage requirements (like waking a device from sleep, power saver mode, and low power state).</li>  
<li><a href="#label-in-name">2.5.3 Label in Name</a> — Requires information in a programmatically determinable form; specifically, the programmatic name contains the text of the visual label.</li>

> <li><a href="#target-size-minimum">2.5.8 Target Size (Minimum)</a> — This success criterion uses CSS pixels for defining the target size.  Products with closed functionality may not use CSS pixels as a standard measurement, but the definition of ‘CSS pixel’ still applies as described in <a href="#applying-css-pixel-to-non-web-documents-and-software">Applying “CSS pixel” to Non-Web Documents and Software</a>.  If the system supports a density-independent pixel measurement, it should be used in place of CSS pixels.

Reviewer: ```suggestion
<li><a href="#target-size-minimum">2.5.8 Target Size (Minimum)</a> — This success criterion uses CSS pixels for defining the target size.  ICT with closed functionality may not use CSS pixels as a standard measurement, but the definition of ‘CSS pixel’ still applies as described in <a href="#applying-css-pixel-to-non-web-documents-and-software">Applying “CSS pixel” to Non-Web Documents and Software</a>.  If the system supports a density-independent pixel measurement, it should be used in place of CSS pixels.
```

<div class="note">If the viewing distance and pixel density of the system are unknown, approximating the reference pixel as described in Applying “CSS pixel” to Non-Web Documents and Software is not possible.</div>
<div class="note">For software designed to run on specific known hardware, a physical size standard would be more straightforward to apply, as calculations for a CSS pixel are dependent on the viewing distance or pixel density of the display.</div></li>
<li><a href="#language-of-page">3.1.1 Language of Page</a> — Depends upon language information being in a programmatically determinable form intended to drive correct pronunciation.  Where another mechanism achieves correct pronunciation for products with closed functionality, such as self-voicing, the intent of this success criterion would be met.</li>

> <li><a href="#language-of-parts">3.1.2 Language of Parts</a> — Depends upon language information in a programmatically determinable form intended to drive correct pronunciation.  Where another mechanism achieves correct pronunciation for prodcuts with closed functionality, such as self-voicing, the intent of this success criterion would be met.</li>

Reviewer: ```suggestion
<li><a href="#language-of-parts">3.1.2 Language of Parts</a> — Depends upon language information in a programmatically determinable form intended to drive correct pronunciation.  Where another mechanism achieves correct pronunciation for ICT with closed functionality, such as self-voicing, the intent of this success criterion would be met.</li>
```

<li><a href="#consistent-navigation">3.2.3 Consistent Navigation</a> — This success criterion is interpreted to only apply to “sets of software programs” which are very rare.  See the second note in the section <a href="#applying-sc-3-2-3-consistent-navigation-to-non-web-documents-and-software">Applying SC 3.2.3 Consistent Navigation to Non-Web Documents and Software</a>.</li>
<li><a href="#consistent-identification">3.2.4 Consistent Identification</a> — This success criterion is interpreted to only apply to “sets of software programs” which are very rare.  See the second note in the section <a href="#applying-sc-3-2-4-consistent-identification-to-non-web-documents-and-software">Applying SC 3.2.4 Consistent Identification to Non-Web Documents and Software</a>.</li>

> <li><a href="#error-identification">3.3.1 Error Identification</a> — While it’s important for errors that can be detected to be described to the user, for products with closed functionality, the error description does’t have to be provided in text, as ‘text’ is defined in WCAG 2;</li>

Reviewer: ```suggestion
<li><a href="#error-identification">3.3.1 Error Identification</a> — While it’s important for errors that can be detected to be described to the user, for products with closed functionality, the error description doesn’t have to be provided in text, as ‘text’ is defined in WCAG 2;</li>
```
